src/include/txn_inline.h

```diff
@@ -1631,12 +1631,8 @@ __wt_txn_modify_block(
      */
     if (!rollback && prev_tsp != NULL) {
         if (upd != NULL) {
-            /*
-             * The durable timestamp must be greater than or equal to the commit timestamp unless it
-             * is an in-progress prepared update.
-             */
-            WT_ASSERT(session,
-              upd->durable_ts >= upd->start_ts || upd->prepare_state == WT_PREPARE_INPROGRESS);
+            /* The durable timestamp must be greater than or equal to the commit timestamp. */
+            WT_ASSERT(session, upd->durable_ts >= upd->start_ts);
```

Comment:
The reason this is valid is that the `!rollback` if block wrapping this assertion was added in the past which means the prepare_state cannot be `INPROGRESS` at this point in the code.

Reply:
Thanks. I read through the ticket and earlier ones referenced in it.